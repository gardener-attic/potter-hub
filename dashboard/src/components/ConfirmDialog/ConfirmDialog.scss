.confirm-dialog-loading-info {
  .loading-legend {
    line-height: 40px;
    padding-right: 5px;
  }

  .spinner {
    margin: 0;
  }
}
