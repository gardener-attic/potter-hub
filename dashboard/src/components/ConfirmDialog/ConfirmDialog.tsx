import * as React from "react";
import Modal from "react-modal";
import LoadingWrapper from "../LoadingWrapper";
import "./ConfirmDialog.css";

interface IConfirmDialogProps {
  modalIsOpen: boolean;
  loading: boolean;
  extraElem?: JSX.Element;
  confirmationText?: string;
  confirmationButtonText?: string;
  onConfirm: () => any;
  closeModal: () => any;
}

interface IConfirmDialogState {
  error?: string;
  modalIsOpen: boolean;
}

class ConfirmDialog extends React.Component<IConfirmDialogProps, IConfirmDialogState> {
  public state: IConfirmDialogState = {
    error: undefined,
    modalIsOpen: this.props.modalIsOpen,
  };

  public render() {
    return (
      <div className="ConfirmDialog">
        <Modal
          className="centered-modal"
          isOpen={this.props.modalIsOpen}
          onRequestClose={this.closeModal}
          contentLabel="Modal"
        >
          {this.state.error && (
            <div className="padding-big margin-b-big bg-action">{this.state.error}</div>
          )}
          {this.props.loading === true ? (
            <div className="row confirm-dialog-loading-info">
              <div className="col-8 loading-legend">Loading, please wait</div>
              <div className="col-4">
                <LoadingWrapper />
              </div>
            </div>
          ) : (
            <div>
              <div className="margin-b-normal">
                {this.props.confirmationText || "Are you sure you want to delete this?"}
              </div>
              {this.props.extraElem}
              <div className="margin-t-normal button-row">
                <button className="button" onClick={this.props.closeModal}>
                  Cancel
                </button>
                <button
                  className="button button-primary button-danger"
                  type="submit"
                  onClick={this.props.onConfirm}
                >
                  {this.props.confirmationButtonText || "Delete"}
                </button>
              </div>
            </div>
          )}
        </Modal>
      </div>
    );
  }

  public openModel = () => {
    this.setState({
      modalIsOpen: true,
    });
  };

  public closeModal = () => {
    this.setState({
      modalIsOpen: false,
    });
  };
}

export default ConfirmDialog;
