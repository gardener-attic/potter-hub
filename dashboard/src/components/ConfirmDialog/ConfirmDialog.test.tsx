import { shallow } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";
import itBehavesLike from "../../shared/specs";
import ConfirmDialog from "./ConfirmDialog";

context("when loading is true", () => {
  const props = {
    loading: true,
  };

  itBehavesLike("aLoadingComponent", { component: ConfirmDialog, props });
});

const defaultProps = {
  loading: false,
  modalIsOpen: true,
  onConfirm: jest.fn(),
  closeModal: jest.fn(),
};

it("should modify the default confirmation text", () => {
  const wrapper = shallow(
    <ConfirmDialog {...defaultProps} confirmationText="Sure?" confirmationButtonText="Sure!" />,
  );
  expect(wrapper.find(".margin-b-normal").filterWhere(d => d.text() === "Sure?")).toExist();
  expect(wrapper.find("button").filterWhere(d => d.text() === "Sure!")).toExist();
});
