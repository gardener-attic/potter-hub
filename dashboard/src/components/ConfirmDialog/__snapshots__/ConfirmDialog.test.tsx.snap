// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading is true loading spinner matches the snapshot 1`] = `
<div
  className="ConfirmDialog"
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="centered-modal"
    closeTimeoutMS={0}
    contentLabel="Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="row confirm-dialog-loading-info"
    >
      <div
        className="col-8 loading-legend"
      >
        Loading, please wait
      </div>
      <div
        className="col-4"
      >
        <LoadingWrapper
          loaded={false}
          type={0}
        />
      </div>
    </div>
  </Modal>
</div>
`;
